### PR TITLE
Transform TextureView according to aspect ratio

### PR DESCRIPTION
**Pull request by davidjiagoogle**
Added SurfaceTransformatiomUtil which contains transformation logics borrowed from CameraX's ViewFinderTransformation.

The Texture is scaled to fill center the parent view.

Tested on Samsung, Pixel, and foldable devices

### CODE REVIEW

File: domain/camera/src/main/java/com/google/jetpackcamera/domain/camera/CameraXCameraUseCase.kt [hunk 1]
```diff
@@ -72,18 +72,23 @@ class CameraXCameraUseCase @Inject constructor(
...
     private var recording: Recording? = null
 
     private lateinit var useCaseGroup: UseCaseGroup
 
-    private lateinit var aspectRatio : AspectRatio
+    private lateinit var aspectRatio: AspectRatio
     private var singleStreamCaptureEnabled = false
     private var isFrontFacing = true
 
+    private var currentSurfaceProvider: SurfaceProvider? = null
```
Comment (yasith) [inline domain/camera/src/main/java/com/google/jetpackcamera/domain/camera/CameraXCameraUseCase.kt:90]: Why do we need to hold on to this separately?

File: camera-viewfinder-compose/src/main/java/com/google/jetpackcamera/viewfinder/CameraPreview.kt [hunk 2]
```diff
@@ -41,13 +42,13 @@ import kotlinx.coroutines.flow.mapNotNull
 
 private const val TAG = "Preview"
 
+@SuppressLint("RestrictedApi")
 @Composable
 fun CameraPreview(
-    modifier: Modifier,
```
Comment (yasith) [inline camera-viewfinder-compose/src/main/java/com/google/jetpackcamera/viewfinder/CameraPreview.kt:46]: We need to be able to pass in an optional the modifier for this one. It's a requirement for public Compose APIs

`modifier: Modifier = Modifier,`

**Review by yasith — CHANGES_REQUESTED** [review]

File: camera-viewfinder-compose/src/main/java/com/google/jetpackcamera/viewfinder/surface/Texture.kt [hunk 3]
```diff
@@ -16,87 +16,146 @@
...
     Log.d(TAG, "Texture")
 
+
+    val resolution = surfaceRequest?.resolution
     var textureView: TextureView? by remember { mutableStateOf(null) }
+    var parentView: FrameLayout? by remember { mutableStateOf(null) }
+    if (parentView != null && surfaceRequest != null && resolution != null) {
+        val parentViewSize = Size(parentView!!.width, parentView!!.height)
+        surfaceRequest.setTransformationInfoListener(Dispatchers.Main.asExecutor())
+        { transformationInfo ->
```
[the remark this answers is not in the document]
Comment (davidjiagoogle) [reply camera-viewfinder-compose/src/main/java/com/google/jetpackcamera/viewfinder/surface/Texture.kt:62]: The formatting looks correct to me here

File: domain/camera/src/main/java/com/google/jetpackcamera/domain/camera/CameraXCameraUseCase.kt (hunk 1 above)
Comment (davidjiagoogle) [reply domain/camera/src/main/java/com/google/jetpackcamera/domain/camera/CameraXCameraUseCase.kt:90]: oops I was trying something else. Forgot to revert this

**Pull request closed without merging**